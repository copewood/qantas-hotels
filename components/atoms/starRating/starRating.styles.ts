import styled from 'styled-components';

export const StyledRatingContainer = styled.span`
  width: 120px;
  font-size: 12px;
  margin-left: 1rem;
`;