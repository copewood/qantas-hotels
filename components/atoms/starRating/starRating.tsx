import React, { FC } from 'react';

type StarRatingProps = {};

export const StarRating: FC<StarRatingProps> = ({}) => {
  return <>Star rating</>;
};
