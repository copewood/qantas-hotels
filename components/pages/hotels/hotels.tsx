import React, { FC, useEffect, useState} from 'react';

import { HotelListHeader } from 'components/features/hotelListHeader/hotelListHeader';
import { HotelList } from 'components/features/hotelList/hotelList';
import { Hotel } from '@components/models/hotel';

type HotelsProps = {};

const fetchHotels = async () => {
    return fetch('/api/hotels')
        .then((response) => response.json())
        .then((responseJson) => responseJson);
}

export const Hotels: FC<HotelsProps> = ({}) => {
  const [hotels, setHotels] = useState<Array<Hotel>>();

  useEffect(() => {
    fetchHotels().then((response)=>{
      setHotels(response.results as Array<Hotel>)
    });
  },[])
  
  return <>
    {hotels && <>
      <HotelListHeader hotels={hotels}/>
      {/* <HotelList hotels={hotels}/> */}
    </>}
  </>;
};
