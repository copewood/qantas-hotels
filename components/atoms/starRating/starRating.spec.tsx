import React from 'react';
import { render } from '@testing-library/react';
import { StarRating } from './starRating';

import { mockHotels } from '../../../pages/api/hotels';

describe('HotelList', () => {
  it('should show StarRating', () => {
    const { container } = render(<StarRating rating={mockHotels.results[0].property.rating}/>);
    expect(container).toMatchSnapshot()
  });
});
