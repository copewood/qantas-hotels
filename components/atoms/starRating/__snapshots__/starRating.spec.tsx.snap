// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HotelList should show StarRating 1`] = `
<div>
  <span
    class="sc-bczRLJ dzjrjx"
  >
    Star rating 4.5
  </span>
</div>
`;
